File: unity_app/Assets/Scripts/Core/Enums/ConsequenceEnums.cs
namespace EventPlannerSim.Core
{
    /// <summary>
    /// Types of random events that can occur during event execution.
    /// Requirements: R12.2-R12.10
    /// </summary>
    public enum RandomEventType
    {
        // Vendor Issues
        VendorNoShow,
        VendorLate,
        VendorUnderperformance,
        
        // Equipment Issues
        EquipmentFailure,
        PowerOutage,
        AVMalfunction,
        
        // Guest Issues
        GuestConflict,
        UnexpectedGuests,
        GuestInjury,
        
        // Weather (outdoor events)
        WeatherChange,
        ExtremeWeather,
        
        // Client Issues
        LastMinuteChanges,
        ClientComplaint,
        BudgetDispute,
        
        // Positive Events (rare)
        UnexpectedCompliment,
        MediaCoverage,
        CelebrityAppearance
    }
}

File: EventApp/EventsApp/Models/Enums/ConsequenceEnums.swift
import Foundation

enum RandomEventType: String, CaseIterable, Codable, Hashable {
    case vendorNoShow
    case vendorLate
    case vendorUnderperformance
    case equipmentFailure
    case powerOutage
    case avMalfunction
    case guestConflict
    case unexpectedGuests
    case guestInjury
    case weatherChange
    case extremeWeather
    case lastMinuteChanges
    case clientComplaint
    case budgetDispute
    case unexpectedCompliment
    case mediaCoverage
    case celebrityAppearance
}
